presto-native-execution/presto_cpp/main/PrestoServer.cpp

```diff
@@ -152,6 +153,11 @@ void PrestoServer::run() {
             "Https Client Certificates are not configured correctly");
       }
       clientCertAndKeyPath = optionalClientCertPath.value();
+
+      VELOX_USER_CHECK(
```

Comment:
We can remove this check if we only use `internalCommunicationSharedSecret()`

Reply:
I'm not sure I understand the comment. The check is there to avoid a bad configuration as there are two knobs (because of the java implementation).
Are you suggesting having only on knob in which case we wouldn't need a check to make sure it is properly configured?